listCustomers.php

```diff
@@ -0,0 +1,61 @@
+<?php 
+include("isJson.php");
+include("greatCircleDistance.php");
+
+$intercomLatitude = "53.339428"; 
+$intercomLongitude = "-6.257664";
+$errorMessage = "";
+$listToInvite = array();
+
+$contents = @file_get_contents("customers.json");
+
+if($contents === FALSE){
+	$errorMessage = "Sorry, no such file exists";
+}else{
+	if(isJson($contents)){
+		$recordsArray  = json_decode(trim($contents), true);
+		foreach($recordsArray as $record){
+			if(greatCircleDistance($intercomLatitude, $intercomLongitude, $record["latitude"], $record["longitude"]) < 100000){//100 kms in metres
+					$listToInvite[$record["user_id"]] = $record["name"];
+			}			
+		}
+
+		if(count($listToInvite) > 0){
+			ksort($listToInvite);// sort by key 		
+			include("viewListCustomers.phtml");	
+		}
+		else
+		{
+			$errorMessage = "No customers found";
+		}
+
+	}
+	else
+	{
+		switch (json_last_error()) {
+	        case JSON_ERROR_DEPTH:
+	            $errorMessage = 'JSON Error: Maximum stack depth exceeded';

```

Comment:
It's a bit strange to see all of the JSON  error cases handled here individually. Is there some better way you can just grab a generic error including more details of what actually went wrong ? Because these messages won't necessarily even help you debug what went wrong. E.g. the syntax error, it doesn't say what the syntax error was